src/main/resources/frontend/src/network/ServerCommunicator.ts

```diff
@@ -0,0 +1,269 @@
+import { useAuthStore } from '@/stores/auth'
+import { ServerError } from '@/network/ServerError'
+import { useAppConfigStore } from '@/stores/appConfig'
+
+/**
+ * Utility for making authenticated HTTP requests to the server with automatic error handling
+ * and response parsing.
+ *
+ * @example
+ * // GET request expecting a User response
+ * const user = await ServerCommunicator.getRequest<User>('/api/user');
+ *
+ * // POST request with body, not expecting response
+ * await ServerCommunicator.postRequest<void>('/api/logs', { event: 'action' }, false);
+ */
+export const ServerCommunicator = {
+  getRequest: getRequest,
+  postRequest: postRequest,
+  patchRequest: patchRequest,
+  doUnprocessedRequest: doUnprocessedRequest
+}
+
+/**
+ * Makes a GET request to the specified endpoint.
+ * @template T - The type of the expected response (when expectResponse is true)
+ * @param {string} endpoint - The API endpoint to call
+ * @param {boolean} [expectResponse=true] - Whether to expect and parse a response
+ * @returns {Promise<T | null>} Promise that resolves to:
+ *   - The response data of type T when expectResponse is true
+ *   - null when expectResponse is false
+ * @throws {ServerError} When the request fails (meaning the server returned a code other than 2XX)
+ * @throws {Error} when expectResponse is true but no response is received
+ */
+function getRequest<T>(endpoint: string, expectResponse: false): Promise<null>;
```

Comment:
Note that the generic `<T>` is actually unused on this overload and can be removed.

Reply:
Removed!